test/VoteManager.js

```diff
@@ -652,6 +652,146 @@ describe('VoteManager', function () {
+
+        const epoch = await getEpoch();
+        await stakeManager.connect(signers[8]).stake(epoch, tokenAmount('19000'));
+        await stakeManager.connect(signers[9]).stake(epoch, tokenAmount('17000'));
+
+        const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tree = merkle('keccak256').sync(votes);
+        const root = tree.root();
+        const commitment1 = utils.solidityKeccak256(
+          ['uint256', 'uint256', 'bytes32'],
+          [epoch, root, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+        );
+
+        await voteManager.connect(signers[8]).commit(epoch, commitment1);
+        const stakerIdAcc8 = await stakeManager.stakerIds(signers[8].address);
+        const commitment2 = await voteManager.getCommitment(epoch, stakerIdAcc8);
+
+        assertBNEqual(commitment1, commitment2, 'commitment1, commitment2 not equal');
+      });
+
+      it('if reported value is zero, should not be able to reveal', async function () {
+        const epoch = await getEpoch();
+
+        const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tree = merkle('keccak256').sync(votes);
+
+        await mineToNextState(); // reveal
+
+        const proof = [];
+        for (let i = 0; i < votes.length; i++) {
+          proof.push(tree.getProofPath(i, true, true));
+        }
+
+        const tx = voteManager.connect(signers[8]).reveal(epoch, tree.root(), votes, proof,
+          '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd',
+          signers[8].address);
+
+        await assertRevert(tx, 'revert');
+      });
+      it('if the reported value is zero, should not be able to propose', async function () {
+        const epoch = await getEpoch();
+
+        await mineToNextState(); // propose
+        const stakerIdAcc8 = await stakeManager.stakerIds(signers[8].address);
+        const staker = await stakeManager.getStaker(stakerIdAcc8);
+
+        const { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+        const iteration = await getIteration(voteManager, stakeManager, staker);
+
+        const medians = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tx = blockManager.connect(signers[8]).propose(epoch,
+          [19, 20, 21, 22, 23, 24, 25, 26, 27],
+          medians,
+          iteration,
+          biggestInfluencerId);
+
+        await assertRevert(tx, 'revert');
+      });
+
+      it('if the reported value is zero, should not be able to dispute', async function () {
+        await mineToNextState(); // dispute
+        const epoch = await getEpoch();
+
+        const sortedVotes = [toBigNumber('0')];
+
+        const tx = blockManager.connect(signers[9]).giveSorted(epoch, 1, sortedVotes);
+
+        await assertRevert(tx, 'sorted[i] is not greater than lastVisited');
+      });
+
+      it('if the reported value is zero, only commit should work in next epoch', async function () {
```

Comment:
whats reported value here? revealed/propsoed/something else?

Reply:
This is the next epoch, where I check if the reported values are zero, everything should work as we expected.
In this epoch everything starts  fresh ,to check that penalties should not be applies in this epoch for staker for reporting zero value. As the staker can commit in next epoch after reporting zero value in previous epoch , I thought this was the test case we were mentioning to add here.

Reply:
there is no such thing as reported value. mention what do you mean exactly.  revealed/proposed value